README.md

```diff
@@ -1,2 +1,16 @@
 # merkle_tree
-This is an implementation of a merkle tree in rust.
+This is an implementation of a binary merkle tree in rust.
```

Comment:
This readme is ok, but we need a bit more:
- Explain how to use this tree.
- A little detail into how this implementation works.
- Examples.
- Build comands.
- Add a makefile, so I can simple clone the repo and do make build, make run, make test, etc. and get up to speed.